### ISSUE TITLE
Convenience exec/query overloads that do parameterized SQL

### ISSUE THREAD

**Issue by Abscissa** [mysql-d/mysql-native#118]
...using prepared statements behind-the scenes.

Ie:

`conn.query("SELECT * FROM mytable WHERE id=? AND name=?", 7, "Frank")`

**Comment by schveiguy** [comment]
One other simple possibility is to expand the `query` and `exec` methods of `PreparedImpl` to allow one-line execution of sequences.

For example:

```D
auto prep = conn.query("SELECT * FROM mytable WHERE id=? AND name=?");
foreach(r; prep.query(7, "Frank")) { ... }
```

If you do go for the one liner, one thing that the current implementation makes a bit cumbersome is that you have to store the `Prepared` somewhere while you iterate, or it will try to purge the result and release the prepared statement (this is how I came across #117). The way I solved it is to create a wrapper struct that holds both the `Prepared` struct and the `ResultRange`, and I `alias this` the range. Of course, if you mitigate that in ways that we discussed in that issue, it may make things much nicer.

**Comment by quickfur** [comment]
FWIW, in my NIH version of `sqlite.d`, I just use `RefCounted` on the prepared statement (which in turn holds a reference to a `RefCounted` database handle) so that I don't have to worry about keeping track of who's referencing what, where, and when they will be done. I just let the refcounting do its job. :-P

**Comment by Abscissa** [comment]
FWIW, The refcoutning problem @schveiguy refers to above was fixed in [v1.1.4](https://github.com/mysql-d/mysql-native/blob/master/CHANGELOG.md#v114---2017-12-04) (see second-to-last bullet item under v1.4.4)

**Comment by Abscissa** [comment]
Added in branch "v2.1.x" via 7a8fc46c37f54472d46872bab630ff256504fb62, e793af7aad87769e02d8a788a5e50b189192ec7c, and 35cddc8b46a46f202e89128240ef06ff0d7b161b

**Comment by schveiguy** [comment]
Nice. I may be able to just about eliminate my templates that do this :)

**Comment by schveiguy** [comment]
I have to come back here to say, was able to trim out ALL of my cruft in my DB wrapper. Things in mysql-native are looking awesome!

* I no longer need Refcounted wrapper structs
* I no longer need these convenience wrappers (I simply had to rename my `fetch` calls to `query`, and things worked seamlessly, and I had already named my non-data-returning function `exec`, so no changes needed there).
* I no longer have extra scopes to ensure ref counting does the "right thing". Things just work.

I was able to delete probably about 130 lines of code from my project that just dealt with wrapping and lifetime. Kudos!
